draco/core/containers.py

```diff
@@ -316,6 +316,47 @@ def axes(self):
         # hash randomization
         return tuple(sorted(axes))
 
+    def _make_selections(self, sel_args):
+        """
+        Match down-selection arguments to axes of datasets.
+
+        Parses sel_* argument and returns dict mapping dataset names to selections.
+
+        Parameters
+        ----------
+        sel_args : dict
+            Should contain valid numpy indexes as values and axis names (str) as keys.
+
+        Returns
+        -------
+        dict
+            Mapping of dataset names to numpy indexes for downselection of the data.
+            Also includes another dict under the key "index_map" that includes
+            the selections for those.
+        """
+        # Check if all those axes exist
+        for axis in sel_args.keys():
+            if axis not in self._axes:
+                raise RuntimeError("No '{}' axis found to select from.".format(axis))
+
+        # Build selections dict
+        selections = {}
+        for name, dataset in self._dataset_spec.items():
```

Comment:
This will insert dataset selection entries even when they have no relevant axes. Should you just leave them missing?